Month3/Project1/Project1/TDinamicArray.h

```diff
@@ -0,0 +1,290 @@
+	{
+		return Array[Size];
+	}
+
+
+	T& GetData()
+	{
+		return Array;
+	}
+	const T& GetData() const
+	{
+		return Array;
+	}
+
+	bool Empty()
+	{
+		return Capacity == 0;
+	}
+
+	int GetSize()
+	{
+		return Size;
+	}
+
+	int GetCapacity()
+	{
+		return Capacity;
+	}
+
+
+	void Reserve(int RequieredCapacity)
+	{
+		if (RequieredCapacity > Capacity)
+		{
+			T* NewArray = new T[RequieredCapacity];
+			for (int CurrentIndex = 0; CurrentIndex < Capacity; CurrentIndex++)
+			{
+				NewArray[CurrentIndex] = Array[CurrentIndex];
+			}
+			delete Array;
+
+			Capacity = RequieredCapacity;
+			Array = NewArray;
+		}
+
+		
+	}
+
+
+	void ShriktoFit()
+	{
+
+		T* NewArray = new T[Size];
+
+		for (int CurrentIndex = 0; CurrentIndex <= Size; CurrentIndex++)
+		{
+			NewArray[CurrentIndex] = Array[CurrentIndex];
+		}
+		delete Array;
+
+		Capacity = Size;
+		Array = NewArray;
+	}
+
+	void Clear()
+	{
+		if (Size > 0)
+		{
+			Size = 0;
+		}
+	}
+
+
+	void PushBack(const T& NewShape)
+	{
+		if (Capacity == 0)
+		{
+			Reserve(1);
+			Size++;
+			Array[Size] = NewShape;
+		}
+		if (Size == Capacity)
+		{
+			int NewCapacity = Capacity * 2;
+			T* NewArray = new T[NewCapacity];
+			for (int CurrentIndex = 0; CurrentIndex <= Capacity; CurrentIndex++)
+			{
+				NewArray[CurrentIndex] = Array[CurrentIndex];
+			}
+			Capacity = NewCapacity;
+			delete Array;
+			Array = NewArray;
+			Size += 1;
+			Array[Size] = NewShape;
+		}
+	}
+
+
+	void PopBack()
+	{
+		Size -= 1;
+	}
+
+	void Insert(int Index, const T& Shape)
+	{
+		std::cout << "In which position do you want to add your Shape?" << std::endl;
+		std::cin >> Index;
+		if (Capacity == Index)
+		{
+			int NewCapacity = Capacity * 2;
+			T* NewArray = new T[NewArray];
+			for (int CurrentIndex = 0; CurrentIndex <= Capacity; CurrentIndex++)
+			{
+				NewArray[CurrentIndex] = Array[CurrentIndex];
+			}
+			delete Array;
+
+			Capacity = NewCapacity;
+			Array = NewArray;
+			if (Size == Index)
+			{
+				Array[Size + 1] = Array[Size];
+				Array[Index] = Shape;
+			}
+			Size += 1;
+
+		}
+		if (Capacity == Size)
+		{
+			int NewCapacity = Capacity * 2;
+			T* NewArray = new T[NewArray];
+			for (int CurrentIndex = 0; CurrentIndex <= Capacity; CurrentIndex++)
+			{
+				NewArray[CurrentIndex] = Array[CurrentIndex];
+			}
+			delete Array;
+
+			Capacity = NewCapacity;
+			Array = NewArray;
+			for (int CurrentIndex = Size; CurrentIndex >= Index; CurrentIndex--)
+			{
+				Array[CurrentIndex + 1] = Array[CurrentIndex];
+			}
+			Size++;
+			Array[Index] = Shape;
+		}
+
+	}
+
+	void Erase(int Index)
+	{
+		std::cout << "In which position do you want to delete the Shape?" << std::endl;
+		std::cin >> Index;
+		if (Index < Size && Size > 1)
+		{
+			for (int CurrentIndex = Index; CurrentIndex < Size - 1; CurrentIndex++)
+			{
+				Array[CurrentIndex] = Array[CurrentIndex + 1];
+
+			}
+		}
+		if (Size == Capacity && Size == 1)
+		{
+			Size = 0;
+		}
+		if (Index == Size)
+		{
+			Size -= 1;
+		}
```

Comment:
I think in either case, if you are erasing something, the size should decrease. If you don't want this to interfere with the previous if clause, you can use an else clause.